src/main/java/seedu/address/model/Model.java

```diff
@@ -70,6 +71,12 @@
      */
     void addPerson(Person person);
 
+    void addMod(Mod mod);
+
+    Optional<Mod> getMod(Mod mod);
+
+    String getDefaultGroupModel(Mod mod);
```

Comment:
This method is a little oddly named in my opinion, maybe something like getDefaultGroupOfMod() would be better?

Similarly for isDefaultPresent(Mod) and retreievePreviousDefault(Mod)

These methods do not indicate that they are doing operations in relation to mods